processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -0,0 +1,212 @@
+/**
+ *  Copyright 2012-2017 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.internal.model;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.mapstruct.ap.internal.model.common.Parameter;
+import org.mapstruct.ap.internal.model.source.Mapping;
+import org.mapstruct.ap.internal.model.source.MappingOptions;
+import org.mapstruct.ap.internal.model.source.Method;
+import org.mapstruct.ap.internal.model.source.PropertyEntry;
+import org.mapstruct.ap.internal.model.source.SourceReference;
+
+import static org.mapstruct.ap.internal.util.Collections.first;
+
+/**
+ * This is a helper class that holds the generated {@link PropertyMapping}(s) and all the information associated with
+ * it for nested target properties.
+ *
+ * @author Filip Hrisafov
+ */
+public class NestedTargetPropertyMappingHolder {
```

Comment:
Is it only nested target properties..? Or is it handling defined mappings in general?